### PR TITLE
Search Dashboard: Updating PlanUsageSection to more closely match design

### LINKED ISSUE #26761: Search: Update donut meter for customers on old paid plans
For customers on old paid plans, show the requests limit as "Unlimited" with a filled green donut meter.

Before:

> image

After:

> image

### LINKED ISSUE #26781: PlanInfo: Remove "View details" links for now.
The design called for one "View details" toggle and it was meant to show/hide the Record Meter UI. That is on hold for the initial release and we can revisit this after we ship.

Image

### LINKED ISSUE #26782: PlanInfo: Prevent wrapping of icon/details.
The icons and "View details" UI bits should not wrap.

Image

### PR DESCRIPTION
**Pull request by a8ck3n**
Fixes #26781 

#### Changes proposed in this Pull Request:

- Adds callbacks to the UsageMeters component.
- Adds props to DonutMeterContainer so we can pass in the callbacks.
- Fixes a wrapping issue in the DonutMeterContainer.
- Removes the "View details" link that was initially meant to show/hide the Record Meter bar.
- Adds icon indicating external support link.
- Adds special case for handling "unlimited" requests.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

1. Visit any site with a Jetpack Search plan.
2. Open the Search dashboard and append `&new_pricing_202208=1` to the URL.
3. Confirm the meters show correctly with the "i" icons.
4. Confirm the "View details" link is no longer present.

Before:
WrapBefore

After:
After2

### CODE REVIEW

**Comment by jsnmoon** [comment]
Merged the [latest trunk](https://github.com/Automattic/jetpack/pull/26783/commits/f014d30308f40a54379e96b43e01c0052b16415c) and pushed [two](https://github.com/Automattic/jetpack/pull/26783/commits/79e99049104fbde79f89e74fbe84bfebab03be2f) [tweaks](https://github.com/Automattic/jetpack/pull/26783/commits/9f746b11a5fed6c6b7330c945e23df8c17dcd8a3).

File: projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx [hunk 1]
```diff
@@ -188,6 +201,14 @@ const AboutPlanLimits = () => {
...
 						/>
 					),
+					jpExternalIcon: (
+						<ExternalLink
+							href="https://jetpack.com/support/search/"
+							rel="noopener noreferrer"
+							target="_blank"
+							className="support-link"
+						/>
+					),
```
Comment (jsnmoon) [inline projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx:211]: Could this be combined with `jpPlanLimits`? There's a bit of code duplication here.

File: projects/packages/search/src/dashboard/components/donut-meter-container/index.jsx [hunk 2]
```diff
@@ -19,22 +19,31 @@ const formatNumberWithSeparators = x => {
...
+	title,
+	tooltip,
+	iconClickedCallback,
+	linkClickedCallback,
+} ) => {
+	// Special case for "unlimited" requests, which has a limit of 64-bit PHP_INT_MAX.
+	const isUnlimitedRequests = limit > 1e18;
+	const localizedUnlimited = __( 'Unlimited', 'jetpack-search-pkg' );
+	const usageInfo = isUnlimitedRequests
+		? `0/${ localizedUnlimited }`
```
Comment (kangzj) [inline projects/packages/search/src/dashboard/components/donut-meter-container/index.jsx:43]: I guess showing the current requests for the previous month wouldn't hurt and will give the user a sense of the new pricing. `0` doesn't really make a lot of sense here IMO.

**Review by jsnmoon — APPROVED** [review]
Looks good 🚢

File: projects/packages/search/src/dashboard/components/donut-meter-container/index.jsx (hunk 2 above)
Comment (a8ck3n) [reply projects/packages/search/src/dashboard/components/donut-meter-container/index.jsx:43]: Good call. I'll have a look at this today.

File: projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx (hunk 1 above)
Comment (a8ck3n) [reply projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx:211]: This was the only way I could figure to do it to get the link the way I wanted it. If I combine them, the link text would be either:

1. "record indexing and request limits.(icon)" or…
2. "record indexing and request limits(icon)" with the period after the icon.

Or we drop the period. Looking at the designs again, I'm seeing we do not specify a period there so I'll give that a try.

**Pull request merged**